File: 1362-closest-divisors/1362-closest-divisors.js
/**
 * @param {number} num
 * @return {number[]}
 */
var closestDivisors = function(num) {
    for (let i = Math.floor(Math.sqrt(num + 2)); i >= 2; i--) {
        if ((num + 1) % i == 0) return [i, (num + 1) / i];
        else if ((num + 2) % i == 0) return [i, (num + 2) / i];
    }
    return [1, num + 1];   
};

File: 1362-closest-divisors/1362-closest-divisors.java
class Solution {
    public int[] closestDivisors(int num) {
        for (int i = (int) Math.sqrt(num + 2); i >= 2; i--) {
            if ((num + 1) % i == 0) return new int[] {i, (num + 1) / i};
            else if ((num + 2) % i == 0) return new int[] {i, (num + 2) / i};
        }
        return new int[] {1, num + 1};        
    }
}